presubmit: Fails when in a symlinked directory

`make presubmit` fails, when the `syzkaller` / current working directory is symlinked.

```sh
cd $(mktemp -d)
ln -s $GOPATH/src/github.com/google/syzkaller .
cd syzkaller
make presubmit
```

This can be fixed with `cd $(realpath $PWD)`, but it seems that this is something that should "just work".

It's trying to look at other things in `gopath` by assuiming 

## Environment Setup

```
$ export GOPATH=$HOME/gopath
$ export PATH="$GOPATH/bin:$PATH"
$ pwd
/usr/local/google/home/riggle/syzkaller
$ realpath $PWD
/usr/local/google/home/riggle/gopath/src/github.com/google/syzkaller
$ git reset --hard origin/master
HEAD is now at 95b5c82b pkg/build: do not use asan for fuchsia
```

## make presubmit

```
$ make presubmit
...
make format
make[2]: Entering directory '/usr/local/google/home/riggle/gopath/src/github.com/google/syzkaller'
export HOSTOS=linux
export HOSTARCH=amd64
export TARGETOS=linux
export TARGETARCH=amd64
export TARGETVMARCH=amd64
export CC=gcc
export ADDCFLAGS=-m64 -static
export NCORES=56
export EXE=

go fmt ./...
clang-format --style=file -i executor/*.cc executor/*.h tools/kcovtrace/*.c
go build "-ldflags=-s -w -X github.com/google/syzkaller/sys.GitRevision=95b5c82b28c3107383df2cd38d9af8c7984bd31c+" -o bin/syz-fmt ./tools/syz-fmt
bin/syz-fmt sys/test
bin/syz-fmt sys/akaros
bin/syz-fmt sys/freebsd
bin/syz-fmt sys/netbsd
bin/syz-fmt sys/linux
bin/syz-fmt sys/fuchsia
bin/syz-fmt sys/windows
stat ../gopath/src/github.com/google/syzkaller/dashboard/app/app_test.go: no such file or directory
stat ../gopath/src/github.com/google/syzkaller/dashboard/app/reporting_email.go: no such file or directory
exit status 2
exit status 2
stat ../gopath/src/github.com/google/syzkaller/dashboard/app/handler.go: no such file or directory
exit status 2
stat ../gopath/src/github.com/google/syzkaller/dashboard/app/reporting.go: no such file or directory
stat ../gopath/src/github.com/google/syzkaller/pkg/csource/build.go: no such file or directory
...
Makefile:169: recipe for target 'format_go' failed
make[2]: *** [format_go] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/usr/local/google/home/riggle/gopath/src/github.com/google/syzkaller'
Makefile:155: recipe for target 'generate' failed
make[1]: *** [generate] Error 2
make[1]: Leaving directory '/usr/local/google/home/riggle/gopath/src/github.com/google/syzkaller'
Makefile:254: recipe for target 'presubmit' failed
make: *** [presubmit] Error 2
```

## cd $(realpath .) && make presubmit

```
$ make presubmit
...
make format
make[2]: Entering directory '/usr/local/google/home/riggle/gopath/src/github.com/google/syzkaller'
export HOSTOS=linux
export HOSTARCH=amd64
export TARGETOS=linux
export TARGETARCH=amd64
export TARGETVMARCH=amd64
export CC=gcc
export ADDCFLAGS=-m64 -static
export NCORES=56
export EXE=

go fmt ./...
clang-format --style=file -i executor/*.cc executor/*.h tools/kcovtrace/*.c
go build "-ldflags=-s -w -X github.com/google/syzkaller/sys.GitRevision=95b5c82b28c3107383df2cd38d9af8c7984bd31c+" -o bin/syz-fmt ./tools/syz-fmt
bin/syz-fmt sys/test
bin/syz-fmt sys/akaros
bin/syz-fmt sys/freebsd
bin/syz-fmt sys/netbsd
bin/syz-fmt sys/linux
pkg/build/linux_generated.go
pkg/build/akaros.go
bin/syz-fmt sys/fuchsia
bin/syz-fmt sys/windows
pkg/vcs/vcs_test.go
prog/prog_test.go
pkg/ifuzz/generated/insns.go

--- a/Makefile
+++ b/Makefile
@@ -166,8 +166,8 @@ bin/syz-sysgen:
 format: format_go format_cpp format_sys
 
 format_go:
-	# When working from a symlinked directory, 'go fmt ./...' will fail.
-	# We need to invoke it from the "real" path inside GOROOT.
+	# When working from a symlinked directory outside of GOPATH, 'go fmt' will fail.
+	# We need to invoke it from a path inside of GOPATH.
 	cd $(realpath ${PWD}) && $(GO) fmt ./...
 
 format_cpp:

--- a/Makefile
+++ b/Makefile
@@ -166,7 +166,9 @@ bin/syz-sysgen:
 format: format_go format_cpp format_sys
 
 format_go:
-	$(GO) fmt ./...
+	# When working from a symlinked directory, 'go fmt ./...' will fail.
+	# We need to invoke it from the "real" path inside GOROOT.
+	cd $(realpath ${PWD}) && $(GO) fmt ./...
 
 format_cpp:
 	clang-format --style=file -i executor/*.cc executor/*.h tools/kcovtrace/*.c

